### DISCUSSION TITLE
The 'no-drop' & 'all-scroll' cursors are unused in Firefox

### DISCUSSION THREAD

**Discussion by Aleksandra0056** [black7375/Firefox-UI-Fix#746] [General]
### Description:
- I want to say, that 'no-drop' & 'all-scroll' cursors are not showing as intended (They are basically unused in all cases).
- I don't want to make a new issue in this repository, so I created rather a new discussion.
- This was unfixed since Windows XP, and all versions from Firefox v1.0 (2003) to Firefox v117 (2023) are affected.
- Please, tell me if it can be fixed by using pure CSS (but I think, that probably yes).
- Yes, I know, that the BugZilla team should fix this firstly, but they didn't fixed it since 2003, so I reported it rather here.

### Actual behaviour:
- The 'no-drop' cursor glyph is the same as 'not-allowed' cursor in all cases.
- The 'all-scroll' cursor glyph is the same as 'move' cursor in all cases.

### Expected behaviour:
- The 'no-drop' cursor would show up if I grab e.g. link and hover over an area, which doesn't allow being link placed (e.g. Firefox navbar).
- The 'all-scroll' cursor would show up if some area in website/page is indicating the ability to scroll something in any direction.

### Definitions:
- 'no-drop' means, that “An item may not be dropped at the current location”.
- While the 'not-allowed' means, that “The requested action will not be carried out”.
- 'all-scroll' means, that “Something can be scrolled in any direction (panned)”.
- While the 'move' means, that “Something is to be moved.”.

### References:
- https://developer.mozilla.org/en-US/docs/Web/CSS/cursor
- https://www.w3.org/TR/css-ui-3/#valdef-cursor-no-drop
- https://www.w3.org/TR/css-ui-3/#valdef-cursor-all-scroll

### This same bug in BugZilla:
- My report (which is clone of 275173 & 275174): https://bugzilla.mozilla.org/show_bug.cgi?id=1843771
- Original report n. 1: https://bugzilla.mozilla.org/show_bug.cgi?id=275173
- Original report n. 2: https://bugzilla.mozilla.org/show_bug.cgi?id=275174
- (I cloned these two original reports in Bugzilla mainly because these issues were reported 19 years ago, which is very obsolete)

### Images:

CSS3 'no-drop' cursor:

![no-drop](https://github.com/black7375/Firefox-UI-Fix/assets/69351811/62102742-d62a-4789-9db2-2db7d846c6e8)

CSS3 'all-scroll' cursor:

![all-scroll](https://github.com/black7375/Firefox-UI-Fix/assets/69351811/38161697-2f50-42d1-a9d7-f07a15cca413)

Archive with original (.cur) cursor files:
https://www.mediafire.com/file/mumzuntzg3xh9a7/Unused_cursors.7z/file

### Other details:
- This is a problem even in Chromium-based browsers…
- Report by me n. 1 in Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1383867
- Report by me n. 2 in Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1383875

**Comment by Aleksandra0056** [comment]
Mr. Black7375, should I create rather new issue? Or should I keep it here, because I don't know.

EDIT: Since I added a lot of details, then I think, that I'll keep it here. Did I good decision?

**Comment by black7375** [comment]
You may have used it for the discussion, not for the issue, because you roughly predicted it, I also think it is a difficult range to solve in this project.
It is difficult to solve in the front because it is necessary to approach the OS and bring information.
